src/lib.rs

```diff
@@ -313,47 +341,55 @@ implement_emulator!(
     doc = "Create an ARM emulator instance for the specified hardware mode.",
     CpuARM,
     Arch::ARM,
-    RegisterARM
+    RegisterARM,
+    stack_pointer => RegisterARM::SP,
+    program_counter => RegisterARM::PC,
+    accumulator => RegisterARM::R0,
 );
 
 implement_emulator!(
     doc = "An ARM64 emulator instance.",
     doc = "Create an ARM64 emulator instance for the specified hardware mode.",
     CpuARM64,
     Arch::ARM64,
-    RegisterARM64
+    RegisterARM64,
 );
 
 implement_emulator!(
     doc = "A M68K emulator instance.",
     doc = "Create a M68K emulator instance for the specified hardware mode.",
     CpuM68K,
     Arch::M68K,
-    RegisterM68K
+    RegisterM68K,
 );
 
 implement_emulator!(
     doc = "A MIPS emulator instance.",
     doc = "Create an MIPS emulator instance for the specified hardware mode.",
     CpuMIPS,
     Arch::MIPS,
-    RegisterMIPS
+    RegisterMIPS,
 );
 
 implement_emulator!(
     doc = "A SPARC emulator instance.",
     doc = "Create a SPARC emulator instance for the specified hardware mode.",
     CpuSPARC,
     Arch::SPARC,
-    RegisterSPARC
+    RegisterSPARC,
 );
 
 implement_emulator!(
     doc = "An X86 emulator instance.",
     doc = "Create an X86 emulator instance for the specified hardware mode.",
     CpuX86,
     Arch::X86,
-    RegisterX86
+    RegisterX86,
+    // special registers
+    // TODO: handle modes
```

Comment:
Is this TODO for something you wanted to add in this pull request? If not I suggest putting a bit more details in the message.